app/utils/service_client/middleware/middleware_base.rb

```diff
@@ -0,0 +1,36 @@
+module ServiceClient
+  module Middleware
+
+    # Base class for middlewares.
+    #
+    # Default implementations for `enter`, `leave` and `error`
+    # return the context object `ctx` without manipulating it.
+    # The subclasses can override all or some of these methods.
```

Comment:
We could of course change the runner so that it always calls enter, leave and error method, without checking. That would work, if all the middleware inherit the base class. Or do you prefer keeping the runner implementation as it is now?